Tests/Runner/DataLoaderTest.php

```diff
@@ -569,4 +570,52 @@ private static function assertDataType($metadata, $expectedType): void
         }
         self::fail('Metadata key ' . Common::KBC_METADATA_KEY_TYPE . ' not found');
     }
+
+    public function testWorkspaceCleanupFailure(): void
+    {
+        $component = new Component([
+            'id' => 'keboola.runner-workspace-test',
+            'data' => [
+                'definition' => [
+                    'type' => 'aws-ecr',
+                    'uri' => '147946154733.dkr.ecr.us-east-1.amazonaws.com/developer-portal-v2/keboola.runner-workspace-test',
+                    'tag' => '1.6.2',
+                ],
+                'staging-storage' => [
+                    'input' => 'workspace-snowflake',
+                    'output' => 'workspace-snowflake',
+                ],
+            ],
+        ]);
+        $clientMock = $this->createMock(Client::class);
+        $clientMock->method('verifyToken')->willReturn($this->clientWrapper->getBasicClient()->verifyToken());
+
+        // exception is not thrown outside
+        $clientMock->expects(self::once())->method('apiPost')->willThrowException(
+            new ClientException('boo')
+        );
+
+        $clientWrapperMock = $this->createMock(ClientWrapper::class);
+        $clientWrapperMock->method('getBasicClient')->willReturn($clientMock);
+        $clientWrapperMock->method('getBranchClientIfAvailable')->willReturn($clientMock);
+
+        $configuration = new Configuration();
+        $configuration->setName('testWorkspaceCleanup');
+        $configuration->setComponentId('keboola.runner-workspace-test');
+        $configuration->setConfiguration([]);
+        $componentsApi = new Components($this->clientWrapper->getBasicClient());
+        $configId = $componentsApi->addConfiguration($configuration)['id'];
+
+        $logger = new TestLogger();
+        $dataLoader = new DataLoader(
+            $clientWrapperMock,
+            $logger,
+            $this->workingDir->getDataDir(),
+            new JobDefinition([], $component, $configId),
+            new OutputFilter(10000)
+        );
+        $dataLoader->cleanWorkspace();
+        $componentsApi->deleteConfiguration('keboola.runner-workspace-test', $configId);
+        self::assertTrue($logger->hasErrorThatContains('Failed to cleanup workspace: boo'));
```

Comment:
Jen dorba poznamecka - ten assert bych prohodil o radek vys, at je jasny, ze to ten `cleanWorkspace` nahlasi error, ne ten `deleteConfiguration`